src/backend/distributed/commands/alter_table.c

```diff
@@ -1572,3 +1577,67 @@ ExecuteQueryViaSPI(char *query, int SPIOK)
 		ereport(ERROR, (errmsg("could not finish SPI connection")));
 	}
 }
+
+
+/*
+ * SwitchToSequentialAndLocalExecutionIfRelationNameTooLong generates the longest shard name
+ * on the shards of a distributed table, and if exceeds the limit switches to sequential and
+ * local execution to prevent self-deadlocks.
+ *
+ * In case of a RENAME, the relation name parameter should store the new table name, so
+ * that the function can generate shard names of the renamed relations
+ */
+void
+SwitchToSequentialAndLocalExecutionIfRelationNameTooLong(Oid relationId,
+														 char *relationName)
+{
+	if (!IsCitusTable(relationId))
+	{
+		return;
+	}
+
+	char *longestShardName = NULL;
+	if (PartitionedTable(relationId))
+	{
+		/* we may not have any partitions yet, and the function will return NULL in that case */
+		longestShardName = LongestPartitionName(relationId);
```

Comment:
I find it confusing to use `longestShardName ` as the output of `LongestPartitionName `, which returns a table. Can we use an intermediate variable like `longestRelationName` or such? Or, even an helper function which calculates `longestShardName ` would be easier to follow, where we still use the intermediate variables to distinguish shards vs tables

Reply:
I ended up with `GetLongestShardName` and `PartitionWithLongestNameRelationId`

Does it look good?